tests/bootstrap_tests/test_configuration_writer.py

```diff
@@ -478,6 +478,34 @@ def test_write_project_sandbox_config(self):
         )
 
 
+class TestInstallationLocationFile(ShotgunTestBase):
+    def test_character_escaping(self):
+        """
+        Ensure that the ' characte is properly escaped
+        when writing out install_location.yml
+        """
+        new_config_root = os.path.join(
+            self.tank_temp, self.short_test_name, "O'Connell"
+        )
+
+        writer = ConfigurationWriter(
+            ShotgunPath.from_current_os_path(new_config_root), self.mockgun
+        )
+
+        install_location_path = os.path.join(
+            new_config_root, "config", "core", "install_location.yml"
+        )
+
+        os.makedirs(os.path.dirname(install_location_path))
+
+        writer.write_install_location_file()
+
+        with open(install_location_path, "rt") as f:
+            paths = yaml.safe_load(f)
+            path = ShotgunPath(paths["Windows"], paths["Linux"], paths["Darwin"])
+        assert path.current_os == new_config_root
```

Comment:
Should it use self.assertEqual instead?

Reply:
We can stop using `assertEqual`, since `pytest` is now our test runner. Errors are more easily debuggable using `assert` when running with `pytest`.